pkg/generator/fluentd/output/syslog/syslog.go

```diff
@@ -0,0 +1,395 @@
+	}
+	return "Unknown Rfc"
+}
+
+func AppName(s *logging.Syslog, matchtags string) Element {
+	if s == nil {
+		return Nil
+	}
+	appname := "appname"
+	if matchtags == source.JournalTags && Rfc(s) == "rfc5424" {
+		return KV(appname, "${$.systemd.u.SYSLOG_IDENTIFIER}")
+	}
+	if s.AppName == "" {
+		return Nil
+	}
+	if IsKeyExpr(s.AppName) {
+		return KV(appname, fmt.Sprintf("${%s}", s.AppName))
+	}
+	if IsTagExpr(s.AppName) {
+		return KV(appname, s.AppName)
+	}
+	if s.AppName == "tag" {
+		return KV(appname, "${tag}")
+	}
+	return KV(appname, s.AppName)
+}
+
+func Tag(s *logging.Syslog, matchtags string) Element {
+	if s == nil {
+		return Nil
+	}
+	program := "program"
+	if matchtags == source.JournalTags && Rfc(s) == "rfc3164" {
+		return KV(program, "${$.systemd.u.SYSLOG_IDENTIFIER}")
+	}
+	if s.Tag == "" {
+		return Nil
+	}
+	if IsKeyExpr(s.Tag) {
+		return KV(program, fmt.Sprintf("${%s}", s.Tag))
+	}
+	if IsTagExpr(s.Tag) {
+		return KV(program, s.Tag)
+	}
+	if s.Tag == "tag" {
+		return KV(program, "${tag}")
+	}
+	return KV(program, s.Tag)
+}
+
+func MsgID(s *logging.Syslog) Element {
+	if s == nil || s.MsgID == "" {
+		return Nil
+	}
+	msgid := "msgid"
+	if IsKeyExpr(s.MsgID) {
+		return KV(msgid, fmt.Sprintf("${%s}", s.MsgID))
+	}
+	return KV(msgid, s.MsgID)
+}
+
+func ProcID(s *logging.Syslog) Element {
+	if s == nil || s.ProcID == "" {
+		return Nil
+	}
+	procid := "procid"
+	if IsKeyExpr(s.ProcID) {
+		return KV(procid, fmt.Sprintf("${%s}", s.ProcID))
+	}
+	return KV(procid, s.ProcID)
+}
+
+func Protocol(o logging.OutputSpec) string {
+	u, _ := urlhelper.Parse(o.URL)
+	return urlhelper.PlainScheme(u.Scheme)
+}
+
+func PayloadKey(s *logging.Syslog) string {
+	if s != nil {
+		return s.PayloadKey
+	}
+	return ""
+}
+
+func BufferKeys(s *logging.Syslog, matchtags string) []string {
+	if s == nil {
+		return output.NOKEYS
+	}
+	keys := []string{}
+	tagAdded := false
+	if matchtags == source.JournalTags {
+		keys = append(keys, "$.systemd.u.SYSLOG_IDENTIFIER")
+	}
+	if IsKeyExpr(s.Tag) {
+		keys = append(keys, s.Tag)
+	}
+	if IsTagExpr(s.Tag) && !tagAdded {
+		keys = append(keys, "tag")
+		tagAdded = true
+	}
+	if s.Tag == "tag" && !tagAdded {
+		keys = append(keys, "tag")
+		tagAdded = true
+	}
+	if IsKeyExpr(s.AppName) {
+		keys = append(keys, s.AppName)
+	}
+	if IsTagExpr(s.AppName) && !tagAdded {
+		keys = append(keys, "tag")
+		tagAdded = true
+	}
+	if s.AppName == "tag" && !tagAdded {
+		keys = append(keys, "tag")
+	}
+	if IsKeyExpr(s.MsgID) {
+		keys = append(keys, s.MsgID)
+	}
+	if IsKeyExpr(s.ProcID) {
+		keys = append(keys, s.ProcID)
+	}
+	if IsKeyExpr(s.Facility) {
+		keys = append(keys, s.Facility)
+	}
+	if IsKeyExpr(s.Severity) {
+		keys = append(keys, s.Severity)
+	}
+	return keys
+}
+
+func ParseJson(o logging.OutputSpec, op Options) Element {
+	return Filter{
+		MatchTags: "**",
+		Element: ConfLiteral{
+			TemplateName: "syslogParseJson",
+			TemplateStr: `{{define "syslogParseJson" -}}
+@type parse_json_field
```

Comment:
looks like indentation issue

Reply:
it is not an indentation issue. but added a new line in the template first line